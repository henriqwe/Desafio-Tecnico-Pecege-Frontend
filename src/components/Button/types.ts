export type TVariant = 'primary' | 'danger' | 'secondary' | 'success'
