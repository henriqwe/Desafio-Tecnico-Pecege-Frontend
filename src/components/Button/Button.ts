import styled from 'styled-components'

const SButtonBase = styled.button`
  width: 100%;
  padding: 8px;
  display: flex;
  justify-content: center;
  border-radius: 6px;
  cursor: pointer;
  outline: none;
  letter-spacing: inherit;
  font-size: inherit;
  text-align: inherit;
`
export const SButtonPrimary = styled(SButtonBase)`
  background-color: rgb(30 64 175);
  color: white;
`
export const SButtonSecondary = styled(SButtonBase)`
  background: rgb(226 232 240);
  color: black;
`
export const SButtonDanger = styled(SButtonBase)`
  background: rgb(220 38 38);
  color: white;
`
export const SButtonSuccess = styled(SButtonBase)`
  background: rgb(132 204 22);
  color: black;
`
