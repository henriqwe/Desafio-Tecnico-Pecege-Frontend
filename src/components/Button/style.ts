import styled from 'styled-components'

const variantOptions = {
  primary: {
    backgroundColor: '#1E40AF',
    color: 'white'
  },
  danger: {
    backgroundColor: '#dc2626',
    color: 'white'
  },
  success: {
    backgroundColor: '#84cc16',
    color: 'black'
  },
  secondary: {
    backgroundColor: '#e2e8f0',
    color: 'black'
  }
}

export type TVariant = keyof typeof variantOptions

export const Button = styled.button<{ variant: TVariant }>`
  border: none;
  display: inline-flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  justify-content: space-between;
  align-items: center;
  border-radius: 0.375rem;
  font-weight: 500;
  cursor: pointer;
  ${({ variant }) => variantOptions[variant]}
`
export const ButtonContainer = styled.div`
  display: flex;
  gap: 0.5rem;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  white-space: nowrap;
`
