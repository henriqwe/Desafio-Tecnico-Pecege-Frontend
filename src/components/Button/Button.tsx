import * as S from '&components/Button/style'
import type { TVariant } from '&components/Button/style'
import { ButtonHTMLAttributes } from 'react'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant: TVariant
}

export function Button({ variant, children, ...res }: ButtonProps) {
  return (
    <S.Button {...res} variant={variant}>
      <S.ButtonContainer>{children}</S.ButtonContainer>
    </S.Button>
  )
}
