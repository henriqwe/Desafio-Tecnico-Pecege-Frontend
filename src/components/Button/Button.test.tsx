import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Button } from '&components/Button/Button'

describe('Button', () => {
  it('should render correctly snapshot', async () => {
    const { container, findByTestId } = render(
      <Button variant="danger" data-testid={'button'}>
        teste
      </Button>
    )

    const button = (await findByTestId('button')) as HTMLButtonElement
    const buttonChildren = button.children

    expect(buttonChildren[0].innerHTML).toEqual('teste')
    expect(container).toMatchSnapshot()
  })
})
