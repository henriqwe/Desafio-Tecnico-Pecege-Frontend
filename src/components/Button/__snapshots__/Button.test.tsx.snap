// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly snapshot 1`] = `
<div>
  <button
    class="sc-bcPKhP fTDSWF"
    data-testid="button"
    variant="danger"
  >
    <div
      class="sc-grXZZQ jsbKLm"
    >
      teste
    </div>
  </button>
</div>
`;
